### PR TITLE
ISPN-10230 ISPN-8071 Remove org.infinispan.atomic package

### PR DESCRIPTION
**Pull request by ryanemerson**
https://issues.jboss.org/browse/ISPN-10230
https://issues.jboss.org/browse/ISPN-8071

### CODE REVIEW

**Comment by wburns** [comment]
If we could remove `MergeOnStore` as well, that would be spectacular.

**Comment by ryanemerson** [comment]
I had a feeling I had forgotten something

File: documentation/src/main/asciidoc/upgrading/upgrading-content.adoc [hunk 1]
```diff
@@ -1,5 +1,13 @@
 == Upgrading from 9.4 to 10.0
 
+=== Removed AtomicMap and FineGrainedAtomicMap
+
+AtomicMapLookup, AtomicMap and FineGrainedAtomicMap have been removed. No direct replacement exists.
```
Comment (wburns) [inline documentation/src/main/asciidoc/upgrading/upgrading-content.adoc]: We can probably point them to use the functional methods (CHM.merge or FunctionalMap) to do similar operations instead.

**Comment by danberindei** [comment]
-1 from me, I don't like removing API without deprecation

**Comment by tristantarrant** [comment]
@**Dan** while I understand the sentiment, this has been proven to be highly broken code and I don't want to wait yet another major release before it can be dropped
If it makes you feel any better, maybe @ryanemerson can create a deprecation PR for 9.4.x

**Comment by anistor** [comment]
Please slay it.

**Comment by tristantarrant** [comment]
Gone

**Pull request merged**